external/twitter/src/main/scala/org/apache/spark/streaming/twitter/TwitterUtils.scala

```diff
@@ -40,7 +40,7 @@ object TwitterUtils {
       twitterAuth: Option[Authorization],
       filters: Seq[String] = Nil,
       storageLevel: StorageLevel = StorageLevel.MEMORY_AND_DISK_SER_2
-    ): ReceiverInputDStream[Status] = {
+    ): ReceiverInputDStream[Status] = ssc.withScope {
```

Comment:
Answer to your first question: this is still need because you need the scope ID even if you don't need the name.

2nd question: It's to avoid duplicating the name of the stream. There are a few methods that create `*KafkaInputDStream` for instance.